packages/components/src/DeliveryReclamation.purs

```diff
@@ -0,0 +1,175 @@
+  }
+
+type Self = React.Self Props State
+
+type Props =
+  { subsno    :: Int
+  , userUuid  :: Persona.UUID
+  , onCancel  :: Effect Unit
+  , onLoading :: Effect Unit
+  , onSuccess :: Persona.DeliveryReclamation -> Effect Unit
+  , onError   :: Persona.InvalidDateInput -> Effect Unit
+  }
+
+deliveryReclamation :: Props -> JSX
+deliveryReclamation = make component { initialState, render, didMount }
+
+initialState :: State
+initialState =
+  { publicationDate: Nothing
+  , claim: Nothing
+  , maxPublicationDate: Nothing
+  }
+
+component :: React.Component Props
+component = React.createComponent "DeliveryReclamation"
+
+didMount :: Self -> Effect Unit
+didMount self = do
+  now <- Now.nowDateTime
+  self.setState _ { maxPublicationDate = Just now }
+
+render :: Self -> JSX
+render self@{ state: { publicationDate, claim, maxPublicationDate }} =
+  DOM.div
+    { className: "clearfix delivery-reclamation--container"
+    , children:
+        [ Grid.row_
+           [ DOM.div
+               { className: "col col-11"
+               , children: [ DOM.h3_ [ DOM.text "Delivery Reclamation" ] ]
+               }
+           , DOM.div
+               { className: "col-1 flex delivery-reclamation--close-icon"
+               , children: [ DOM.div { className: "close-icon" } ]
+               , onClick: handler_ self.props.onCancel
+               }
+           , deliveryReclamationForm
+           ]
+        ]
+    }
+  where
+    deliveryReclamationForm =
+      DOM.form
+          { onSubmit: handler preventDefault (\_ -> submitForm publicationDate claim)
+          , children:
+              [ publicationDayInput
+              , claimExtensionInput
+              , claimNewDeliveryInput
+              , DOM.div
+                  { children: [ submitFormButton ]
+                  , className: "mt2 clearfix"
+                  }
+              ]
+          }
+
+    publicationDayInput = dateInput self self.state.publicationDate "Publication date"
+
+    claimExtensionInput =
+      InputField.inputField
+        { type_: InputField.Radio
+        , placeholder: "Placeholder"
+        , name: "claim"
+        , onChange: \newClaim ->
+                      let
+                        updateClaim x = self.setState _ { claim = x }
+                        parsed = read <$> newClaim
+                      in
+                        case parsed of
+                          Just value -> updateClaim value
+                          Nothing    -> updateClaim $ Nothing
+        , value: Just "Extension"
+        , label: "Extension"
+        , validationError: Nothing
+        }
+
+    claimNewDeliveryInput =
+      InputField.inputField
+        { type_: InputField.Radio
+        , placeholder: "Placeholder"
+        , name: "claim"
+        , onChange: \newClaim ->
+                      let
+                        updateClaim x = self.setState _ { claim = x }
+                        parsed = read <$> newClaim
```

Comment:
Instead of using `map` here, and having `parsed :: Maybe (Maybe User.DeliveryReclamationClaim)`, we could use bind to have `parsed :: Maybe User.DeliveryReclamationClaim`:
`parsed = read =<< newClaim`

Reply:
Yes, I started re-reading the Monads chapter only yesterday lol